### PR TITLE
chore: Setup platform library go module

### PR DESCRIPTION
**Pull request by PeterSchafer**
We are re-purposing this repo and therefore cleaning it up a bit.

### CODE REVIEW

File: .circleci/slack-notify-failure.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#! /bin/sh
+
+JOB_NAME="$@"
+
+curl -X POST -H 'Content-Type:application/json' -d '{"attachments": [{"color": "#EE0000", "fallback": "Build Notification: '$CIRCLE_BUILD_URL'", "title": ":warning: homebase-garbage-collector merge failure :warning:", "text": ":x: CircleCI job `'"${JOB_NAME}"'` failed :x:\n'$CIRCLE_BUILD_URL'"}]}' $SLACK_WEBHOOK
```
Comment (Avishagp) [inline .circleci/slack-notify-failure.sh:5]: Noticed the phrasing is different in success and failure, one says `homebase-garbage-collector` and the other `Homebase Garbage Collector`. Is this intentional? Wouldn't it be nicer to have the same phrasing?

File: internal/utils/api_tokens.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+	"fmt"
+
+	"github.com/snyk/go-application-framework/pkg/configuration"
+)
+
+func GetAuthHeader(config configuration.Configuration) string {
+
+	bearerToken := config.GetString(configuration.AUTHENTICATION_BEARER_TOKEN)
+	if len(bearerToken) > 0 {
+		return fmt.Sprintf("Bearer %s", bearerToken)
```
Comment (Avishagp) [inline internal/utils/api_tokens.go:13]: Should `Bearer` be capitalised and `token` not?

File: internal/utils/array.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+func Remove(input map[string]string, key string) map[string]string {
+	found := false
+	key, found = FindKeyCaseInsensitive(input, key)
+	if found {
+		delete(input, key)
+	}
+	return input
+}
+
+// This method determines whether the given key is in the input map, it therefore looks for the exact match and the key in all capital or lower case letters.
```
Comment (Avishagp) [inline internal/utils/array.go:65]: Can we maybe make this explanation clearer/shorter? especially the first sentence

File: internal/utils/cache-dir.go [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+)
+
+func SnykCacheDir() (string, error) {
+	baseDirectory, err := os.UserCacheDir()
+	if err != nil {
+		return "", err
+	}
+
+	snykCacheDir := path.Join(baseDirectory, "snyk")
+	err = os.MkdirAll(snykCacheDir, 0755)
```
Comment (Avishagp) [inline internal/utils/cache-dir.go:15]: What does the number mean?

File: internal/utils/temp-dir.go [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+// Gets the system temp directory and, if it doesn't exist, attempts to create it.
+func systemTempDirectory(debugLogger *log.Logger) (string, error) {
+	tempDir := os.TempDir()
+	// make sure this directory exists
+	debugLogger.Println("system temp directory:", tempDir)
+	_, err := os.Stat(tempDir)
+	if err != nil {
+		debugLogger.Println("system temp directory does not exist... attempting to create it:", tempDir)
+		err = os.MkdirAll(tempDir, 0755)
```
[the remark this answers is not in the document]
Comment (PeterSchafer) [reply internal/utils/temp-dir.go:17]: https://wiki.archlinux.org/title/File_permissions_and_attributes#Numeric_method

File: internal/utils/cache-dir.go (hunk 4 above)
Comment (PeterSchafer) [reply internal/utils/cache-dir.go:15]: https://wiki.archlinux.org/title/File_permissions_and_attributes#Numeric_method

File: internal/utils/temp-dir.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (Avishagp) [reply internal/utils/temp-dir.go:17]: Could we introduce a variable for it?

**Review by Avishagp — APPROVED** [review]

**Pull request merged**